mars/learn/contrib/pytorch/tests/test_dataset.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 1999-2020 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+import mars.tensor as mt
+import mars.dataframe as md
+from mars.utils import lazy_import
+from mars.learn.contrib.pytorch import MarsDataset, RandomSampler, SequentialSampler, SubsetRandomSampler
+
+torch_installed = lazy_import('torch', globals=globals()) is not None
+
+
+@pytest.mark.skipif(not torch_installed, reason='pytorch not installed')
+def test_MarsDataset(setup_cluster):
+    from torch.utils.data import Dataset
+    import numpy as np
+    import pandas as pd
+
+    # Mars tensor
+    data = mt.random.rand(1000, 32, dtype='f4')
+    labels = mt.random.randint(0, 2, (1000, 10), dtype='f4')
+
+    data_verify = data[0].execute().fetch()
+    labels_verify = labels[0].execute().fetch()
+
+    train_dataset = MarsDataset(data, labels)
+    assert isinstance(train_dataset, Dataset)
+    assert (train_dataset[0][0] == data_verify).all()
```

Comment:
To test equality of Mars tensor and ndarray. please use

```python
np.testing.assert_array_equal(arr1, arr2)
```

Reply:
That's great advice!